reminders/plugin_bot.go

```diff
@@ -220,70 +218,33 @@ var cmds = []*commands.YAGCommand{
 				}
 			}
 
-			// Do the actual deletion
-			err = common.GORM.Delete(reminder).Error
-			if err != nil {
-				return nil, err
-			}
-
-			// Check if we should remove the scheduled event
-			currentReminders, err := GetUserReminders(reminder.UserIDInt())
+			// just deleting from database is enough; we need not delete the
+			// scheduled event since the handler will check database
+			_, err = reminder.DeleteG(parsed.Context(), false /* hardDelete */)
 			if err != nil {
 				return nil, err
 			}
 
-			delMsg := fmt.Sprintf("Deleted reminder **#%d**: '%s'", reminder.ID, limitString(reminder.Message))
-
-			// If there is another reminder with the same timestamp, do not remove the scheduled event
-			for _, v := range currentReminders {
-				if v.When == reminder.When {
-					return delMsg, nil
-				}
-			}
-
-			return delMsg, nil
```

Comment:
The comment here implies that the scheduled event to send the reminder will be removed if there exists no reminder with the same timestamp, but this is not the case. The current code simplifies fully to `return delMsg, nil`, so there is no functional change despite appearances.

Per the new comment added above, it is perfectly fine to not delete the scheduled event and just remove the database entry.